### PR TITLE
fix(types): add type definition for typescript.typeCheck

### PR DESCRIPTION
**Pull request by danielroe**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Added type definition for NuxtConfiguration for new `build.typescript` option.

TypeScript validation was failing with new configuration model.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by danielroe** [comment]
I can also provide more precise types from [fork-ts-checker](https://github.com/Realytics/fork-ts-checker-webpack-plugin) docs if that would be helpful, but we'd then need to keep them in sync. I don't think they currently export the Config type, though it is present in the built npm module.

File: packages/config/types/build.d.ts [hunk 1]
```diff
@@ -58,6 +58,9 @@ export interface NuxtConfigurationBuild {
   templates?: any
   terser?: TerserPluginOptions | boolean
   transpile?: (string | RegExp)[]
+  typescript?: {
+    typeCheck?: { [key: string]: string } | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
```
Comment (kevinmarrec) [inline packages/config/types/build.d.ts]:
```suggestion
    typeCheck?: { [key: string]: any} | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

**Comment by kevinmarrec** [comment]
Thanks for  your PR @danielroe !

I suggested a little change. And for the type of the options of ForkTsChecker, there is IMO too many properties to stay in sync, I prefer see if we can import it from somewhere, or see with the maintainer how we could have them.

File: packages/config/types/build.d.ts [hunk 2]
```diff
@@ -58,6 +58,9 @@ export interface NuxtConfigurationBuild {
...
+  }
   useForkTsChecker?: { [key: string]: string } | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
```
Comment (kevinmarrec) [inline packages/config/types/build.d.ts]:
```suggestion
```

`useForkTsChecker` has been removed so it needs to be removed from types.

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

File: packages/config/types/build.d.ts [hunk 3]
```diff
@@ -58,6 +58,8 @@ export interface NuxtConfigurationBuild {
   templates?: any
   terser?: TerserPluginOptions | boolean
   transpile?: (string | RegExp)[]
-  useForkTsChecker?: { [key: string]: string } | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
+  typescript?: {
+    typeCheck?: { [key: string]: any} | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
```
Comment (kevinmarrec) [inline packages/config/types/build.d.ts]:
```suggestion
    typeCheck?: { [key: string]: any } | boolean // TBD - Couldn't find typedefs for the forkTsCheckerWebpackPlugin options
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

File: packages/config/types/build.d.ts (hunk 3 above)
Comment (kevinmarrec) [reply packages/config/types/build.d.ts]: Oops @danielroe , could you accept this last suggestion so I can aproove the PR ? Thanks.

**Review by kevinmarrec — APPROVED** [review]

**Comment by pi0** [comment]
Thanks @danielroe @kevinmarrec :heart:

**Pull request merged**